erpnext/patches/v14_0/discount_accounting_separation.py

```diff
@@ -2,8 +2,10 @@
 
 
 def execute():
-	doc = frappe.get_doc("Accounts Settings")
-	discount_account = doc.enable_discount_accounting
+	data = frappe.db.sql(
+		'select value from tabSingles where doctype="Accounts Settings" and field="enable_discount_accounting"'
+	)
+	discount_account = data and data[0][0] or 0
 	if discount_account:
```

Comment:
This should be casted to integer, else it's always true. @deepeshgarg007 @rahib-hassan